pkg/apis/apps/v1beta1/defaults_test.go

```diff
@@ -459,41 +492,80 @@ func TestSetDefaultStatefulSet(t *testing.T) {
 			enableMaxUnavailablePolicy: true,
 		},
 		{
-			name: "StatefulSetAutoDeletePVC enabled",
+			name: "PVCDeletionPolicy disabled",
 			original: &appsv1beta1.StatefulSet{
 				Spec: appsv1beta1.StatefulSetSpec{
 					Template: defaultTemplate,
 				},
 			},
 			expected: &appsv1beta1.StatefulSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: defaultLabels,
+				},
 				Spec: appsv1beta1.StatefulSetSpec{
 					Replicas:            &defaultReplicas,
+					MinReadySeconds:     int32(0),
 					Template:            defaultTemplate,
 					PodManagementPolicy: appsv1beta1.OrderedReadyPodManagement,
+					Selector: &metav1.LabelSelector{
+						MatchLabels:      map[string]string{"foo": "bar"},
+						MatchExpressions: []metav1.LabelSelectorRequirement{},
+					},
 					UpdateStrategy: appsv1beta1.StatefulSetUpdateStrategy{
 						Type:          appsv1beta1.OnDeleteStatefulSetStrategyType,
 						RollingUpdate: nil,
 					},
 					RevisionHistoryLimit: ptr.To[int32](10),
+				},
+			},
+			disablePVCDeletionPolicy: true,
+		},
+		{
+			name: "PVCDeletionPolicy disabled, scaledown set",
+			original: &appsv1beta1.StatefulSet{
+				Spec: appsv1beta1.StatefulSetSpec{
+					Template: defaultTemplate,
 					PersistentVolumeClaimRetentionPolicy: &appsv1beta1.StatefulSetPersistentVolumeClaimRetentionPolicy{
-						WhenDeleted: appsv1beta1.RetainPersistentVolumeClaimRetentionPolicyType,
-						WhenScaled:  appsv1beta1.RetainPersistentVolumeClaimRetentionPolicyType,
+						WhenScaled: appsv1beta1.DeletePersistentVolumeClaimRetentionPolicyType,
 					},
+				},
+			},
+			expected: &appsv1beta1.StatefulSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: defaultLabels,
+				},
+				Spec: appsv1beta1.StatefulSetSpec{
+					Replicas:        &defaultReplicas,
+					MinReadySeconds: int32(0),
+					Template:        defaultTemplate,
+					PersistentVolumeClaimRetentionPolicy: &appsv1beta1.StatefulSetPersistentVolumeClaimRetentionPolicy{
+						WhenScaled: appsv1beta1.DeletePersistentVolumeClaimRetentionPolicyType,
+					},
+					PodManagementPolicy: appsv1beta1.OrderedReadyPodManagement,
 					Selector: &metav1.LabelSelector{
 						MatchLabels:      map[string]string{"foo": "bar"},
 						MatchExpressions: []metav1.LabelSelectorRequirement{},
 					},
+					UpdateStrategy: appsv1beta1.StatefulSetUpdateStrategy{
+						Type:          appsv1beta1.OnDeleteStatefulSetStrategyType,
+						RollingUpdate: nil,
+					},
+					RevisionHistoryLimit: ptr.To[int32](10),
 				},
 			},
-			enableStatefulSetAutoDelete: true,
+			disablePVCDeletionPolicy: true,
 		},
 	}
 
 	for _, test := range tests {
 		test := test
 		t.Run(test.name, func(t *testing.T) {
+			if test.disablePVCDeletionPolicy {
```

Comment:
Same comment here.

Reply:
Okay, that means that either we'd have enablePVCDeletionPolicy added to most of the test cases, which seems error prone, or we continue testing with the feature gate off, which means when it is removed the tests will change as well.

I would have thought we would want most tests to be for what's feature-locked.

Reply:
I haven't thought about that, in that case, we can probably leave it in this direction, since that will be less changes.

Reply:
sgtm! thx